crates/polars-ops/src/frame/pivot/mod.rs

```diff
@@ -321,7 +321,10 @@ fn pivot_impl_single_column(
                             let name = expr.root_name()?.clone();
                             let mut value_col = value_col.clone();
                             value_col.rename(name);
-                            let tmp_df = value_col.into_frame();
+                            let tmp_df = value_col
+                                .into_frame()
+                                .hstack(pivot_df.get_columns())
+                                .unwrap();
```

Comment:
Don't create `tmp_df` as what it does now.

Let's say we have `pivot_df: A, B, C` and we create `value_col: A` from `pivot_df`. By concatting/hstacking we create:

`tmp_df: A, A, B, C`. Where we just want to pass the context of `pivot_df` to the aggregation. So we can pass  `pivot_df` directly to `expr.evaluate`.